libraries/rush-lib/src/logic/operations/ShardedPhaseOperationPlugin.ts

```diff
@@ -0,0 +1,212 @@
+    const { associatedPhase: phase, associatedProject: project } = operation;
+    const operationSettings: IOperationSettings | undefined = getOperationSettings(phase, project);
+    if (operationSettings?.sharding) {
+      shardableOperations.add(operation);
+    }
+  }
+  const getCustomParameterValuesForPhase: (phase: IPhase) => ReadonlyArray<string> =
+    getCustomParameterValuesByPhase();
+
+  for (const operation of existingOperations) {
+    const { associatedPhase: phase, associatedProject: project } = operation;
+    const operationSettings: IOperationSettings | undefined = getOperationSettings(phase, project);
+    if (phase && project && operationSettings?.sharding && !operation.runner) {
+      const { count: shards, shardScriptConfiguration } = operationSettings.sharding;
+
+      existingOperations.delete(operation);
+
+      const collatorNode: Operation = new Operation({
+        phase,
+        project
+      });
+      existingOperations.add(collatorNode);
+
+      const parentFolderFormat: string =
+        operationSettings.sharding.outputFolderArgument?.parentFolderNameFormat ??
+        `.rush/operations/${TemplateStrings.PHASE_NAME}/shards`;
+
+      // Create a new one to avoid moving this into `beforeCreateOperation`.
+      const operationMetadataManager: OperationMetadataManager = new OperationMetadataManager({
+        phase,
+        rushProject: project,
+        operation
+      });
+      const parentFolder: string = parentFolderFormat.replace(
+        TemplateStrings.PHASE_NAME,
+        operationMetadataManager.logFilenameIdentifier
+      );
+
+      const collatorDisplayName: string = `${getDisplayName(phase, project)} - collate`;
+      const collatorRawScript: string | undefined = getScriptToRun(project, phase.name, undefined);
+
+      if (collatorRawScript === undefined && phase.missingScriptBehavior === 'error') {
+        throw new Error(
+          `The project '${project.packageName}' does not define a '${phase.name}' command in the 'scripts' section of its package.json`
+        );
+      }
+
+      if (collatorRawScript) {
+        const collatorRunner: ShellOperationRunner = new ShellOperationRunner({
+          commandToRun: collatorRawScript,
+          displayName: collatorDisplayName,
+          phase,
+          rushConfiguration,
+          rushProject: project,
+          operationSettings,
+          environment: {
+            ...process.env,
+            [EnvironmentVariableNames.RUSH_SHARD_PARENT_FOLDER]: parentFolder,
+            [EnvironmentVariableNames.RUSH_SHARD_COUNT]: shards.toString()
+          }
+        });
+        collatorNode.runner = collatorRunner;
+      } else {
+        collatorNode.runner = new NullOperationRunner({
+          name: collatorDisplayName,
+          result: OperationStatus.NoOp,
+          silent: phase.missingScriptBehavior === 'silent'
+        });
+      }
+      for (const dependent of operation.consumers) {
```

Comment:
```suggestion
      for (const consumer of operation.consumers) {
```
Consistent terminology. The reason the apis use `consumer` instead of `dependent` is because the difference between `consumer` and `dependency` is much more obvious than between `dependent` and `dependency`.